### PR TITLE
SQS Module

### PR DESCRIPTION
**Pull request by pk0331**
Module for creating following types of queues. 
   1. Standard
   2. Fifo

### CODE REVIEW

File: templates/infrastructure/environment/aws/sqs/terragrunt.hcl [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+}
+
+# These are the variables we have to pass in to use the module specified in the terragrunt configuration above
+inputs = {
+  default_tags = {
+    "argonaut.dev/name" = "{{.Spec.name}}"
+    "argonaut.dev/manager" = "argonaut.dev"
+    "argonaut.dev/type" = "sqs queue"
+    "argonaut.dev/env/${local.env}" = "true"
```
Comment (suryaoruganti) [inline templates/infrastructure/environment/aws/sqs/terragrunt.hcl:36]: change format to 
"argonaut.dev/env" = "${local.env}"

File: modules/aws/sqs/variables.tf [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+variable "kms_master_key_id" {
+  description = "The ID of an AWS-managed customer master key (CMK) for Amazon SQS or a custom CMK"
+  type        = string
+  default     = null
+}
+
+variable "sqs_managed_sse_enabled" {
+  description = "Boolean to enable server-side encryption (SSE) of message content with SQS-owned encryption keys"
+  type        = bool
+  default     = false
```
Comment (suryaoruganti) [inline modules/aws/sqs/variables.tf:101]: true

File: modules/aws/sqs/variables.tf [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+variable "redrive_allow_policy" {
+  description = "The JSON policy to set up the Dead Letter Queue redrive permission, see AWS docs."
+  type        = object({
+    sourceQueueArns   = string
+  })
+  default     = {
+    sourceQueueArns = null
+  }
+}
```
Comment (suryaoruganti) [inline modules/aws/sqs/variables.tf:78]: should expose this as a user input

File: modules/aws/sqs/variables.tf [hunk 4]
```diff
@@ -0,0 +1,108 @@
+variable "create" {
+  description = "Whether to create SQS queue"
+  type        = bool
+  default     = true
+}
+
+variable "name" {
```
Comment (shubhamgrg04) [inline modules/aws/sqs/variables.tf:7]: @suryaoruganti what inputs do we want from users for SQS? I am thinking of getting

name
fifo_queue
max_message_size
message_retention_seconds
visibility_timeout_seconds

redrive_allow_policy

Comment (suryaoruganti) [reply modules/aws/sqs/variables.tf:7]: these params are not in the `terragrunt.hcl` file

Comment (pk0331) [reply modules/aws/sqs/variables.tf:7]: These 3 Parameters already there in variables.tf.
max_message_size
message_retention_seconds
visibility_timeout_seconds

Comment (pk0331) [reply modules/aws/sqs/variables.tf:7]: redrive_allow_policy 
Added and tested

**Review by mrinalwahal — APPROVED** [review]
lgtm 👍

**Pull request merged**